mllib/src/main/scala/org/apache/spark/mllib/util/modelSaveLoad.scala

```diff
@@ -110,7 +110,7 @@ private[mllib] object Loader {
       assert(loadedFields.contains(field.name), s"Unable to parse model data." +
         s"  Expected field with name ${field.name} was missing in loaded schema:" +
         s" ${loadedFields.mkString(", ")}")
-      assert(loadedFields(field.name) == field.dataType,
+      assert(DataType.equalsIgnoreNullability(loadedFields(field.name), field.dataType),
```

Comment:
@mengxr Is it OK to ignore the nullability of ArrayType (containsNull field) and MapType (valueContainsNull field) in this check?